swap/src/protocol/alice/swap.rs

```diff
@@ -81,10 +64,31 @@ pub fn is_encsig_learned(state: &AliceState) -> bool {
     )
 }
 
+pub async fn run(swap: alice::Swap) -> Result<AliceState> {
+    run_until(swap, is_complete).await
+}
+
+pub async fn run_until(
+    swap: alice::Swap,
+    is_target_state: fn(&AliceState) -> bool,
+) -> Result<AliceState> {
+    run_until_internal(
+        swap.state,
+        is_target_state,
+        swap.event_loop_handle,
+        swap.bitcoin_wallet,
+        swap.monero_wallet,
+        swap.config,
+        swap.swap_id,
+        swap.db,
+    )
+    .await
+}
+
 // State machine driver for swap execution
 #[async_recursion]
 #[allow(clippy::too_many_arguments)]
-pub async fn run_until(
+async fn run_until_internal(
```

Comment:
```suggestion
async fn run_until(
```
I understand why you want to use `internal` but it's not a commonly used terminology. Maybe for for the 2 name changes I propose or `_run_until`.